### PR TITLE
Implemented Sort with a lot of functional code

### PR DESCRIPTION
**Pull request by PBandJaimie**
@EJLEE808 @a-nouraldean @Taniel-P 

Implemented <Sort />

-refactored server route to get reviews in helpful order and newest order
-wrote A LOT of functional code in order to sort reviews based on relevance
-created a select dropdown menu

### CODE REVIEW

File: client/src/app.jsx [hunk 1]
```diff
@@ -82,7 +83,7 @@ class App extends React.Component {
   }
 
   componentDidMount() {
-    this.getAllProductInfo('28212')
+    this.getAllProductInfo('29004')
```
Comment (Taniel-P) [inline client/src/app.jsx:86]: Rather than hard code this here, we can set the default state in the constructor and use this.state.mainProductId here.

Comment (PBandJaimie) [reply client/src/app.jsx:86]: oh yeah, definitely! I will change it, but I didn't write this particular code, just changed the number to test some stuff. Will change now!

Comment (PBandJaimie) [reply client/src/app.jsx]: Oh I see you already changed it! i accepted the incoming change and pushed it up!

**Review by Taniel-P — APPROVED** [review]
Good work!

**Pull request merged**